src/renderers/shaders/ShaderChunk/common.glsl

```diff
@@ -4,6 +4,10 @@
 #define LOG2 1.442695
 #define EPSILON 1e-6
 
+#define square(a) a * a
+#define saturate(a) clamp( a, 0.0, 1.0 )
+#define whiteCompliment(a) 1.0 - saturate( a )
```

Comment:
Please put backets around this or one could get the wrong answer if you use an operator with a greater precedence than "-".